### PR TITLE
[DX] Add bin/clean-phpstan.php to verify cleaned up Ignored error pattern

### PR DESCRIPTION
**Pull request by samsonasik**
Based on https://github.com/rectorphp/rector-src/pull/1755#issuecomment-1026566892

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: bin/clean-phpstan.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        return $isFailure ? self::FAILURE : self::SUCCESS;
+    }
+}
+
+$command = new CleanPhpstanCommand();
+
+$application = new Application();
+$application->add($command);
+$application->setDefaultCommand(CommandNaming::classToName($command::class), true);
+$application->run();
```
Comment (TomasVotruba) [inline bin/clean-phpstan.php:112]: :+1:

**Comment by TomasVotruba** [comment]
Thanks :+1:  
It should remove the fixed errors too.

**Comment by samsonasik** [comment]
Do you mean add `--fix` option? For note, that will be multiple use cases:

- remove 1 line without path definition
- remove multiline with only 1 path definition
- remove 1 or more lines from paths collection as the ignored error that can be removed only in 1 path or more via paths config

**Comment by TomasVotruba** [comment]
Yes, that would be great. Not necessary to report in CI as it would add us more work. Just so we have local cleanup script.

It should be part of fix-cs composer script command

**Comment by samsonasik** [comment]
as there is no error in this repo, I tried in symplify repo which output the warning, it seems for one liner message, it needs to detect multi line with ` \e[39;49m\n` as the report show:

```bash
 [WARNING] Ignored error pattern #Method                                        
           Symplify\\Astral\\NodeFinder\\SimpleNodeFinder\:\:findFirstParentByTyp
           e\(\) should return T of PhpParser\\Node\|null but returns           
           class\-string<T of PhpParser\\Node>\|T of PhpParser\\Node# was not  
           matched in reported errors. 
```

that if printed, it goes to:

```bash
 \e[30;43m [WARNING] Ignored error pattern #Method \e[39;49m\n
 \e[30;43m Symplify\\Astral\\NodeFinder\\SimpleNodeFinder\:\:findFirstParentByTyp\e[39;49m\n
 \e[30;43m e\(\) should return T of PhpParser\\Node\|null but returns \e[39;49m\n
 \e[30;43m class\-string<T of PhpParser\\Node\>\|T of PhpParser\\Node# was not \e[39;49m\n
 \e[30;43m matched in reported errors. \e[39;49m\n
 \e[30;43m \e[39;49m '
```

**Comment by samsonasik** [comment]
- [x] implemented for one liner define message 🎉 

Tested in symplify repo : 

```bash
➜  symplify git:(main) ✗ php bin/clean-phpstan.php

 ! [NOTE] Checking 209 files with "yml", "yaml", "neon" suffixes                

                                                                                
 [OK] No errors found                                                           
                                                                                


                                                                                
 [WARNING] Ignored error pattern #Method                                        
           Symplify\\Astral\\NodeFinder\\SimpleNodeFinder\:\:findFirstParentByTyp
           e\(\) should return T of PhpParser\\Node\|null but returns           
           class\-string<T of PhpParser\\Node>\|T of PhpParser\\Node# was not  
           matched in reported errors.                                          
                                                                                


Removing ignored pattern ...
Ignored errors removed
```

```diff
➜  symplify git:(main) ✗ git diff

diff --git a/phpstan.neon b/phpstan.neon
index 0a7932d91..01c42350c 100644
--- a/phpstan.neon
+++ b/phpstan.neon
@@ -350,7 +350,7 @@ parameters:
 
         # known types
         - '#Method Symplify\\Astral\\NodeFinder\\SimpleNodeFinder\:\:findByType\(\) should return array<T of PhpParser\\Node\> but returns array<PhpParser\\Node\>#'
-        - '#Method Symplify\\Astral\\NodeFinder\\SimpleNodeFinder\:\:findFirstParentByType\(\) should return T of PhpParser\\Node\|null but returns class\-string<T of PhpParser\\Node\>\|T of PhpParser\\Node#'
+
```

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

Current functionality is for one liner message ignore removal, I think we can improve for define with path or multi line paths in separate PR, I currently left it with message: 

```bash
There are still ignored errors that need to be cleaned up manually
```

with re-run to display message to clean up.

**Comment by samsonasik** [comment]
rebased.

**Comment by TomasVotruba** [comment]
Thank you! Let's ship it :+1: 

I'll merge this now, but could you add simple test case into `/tests` to verify the error is removed?

**Comment by samsonasik** [comment]
I will try.

**Pull request merged**